pkg/boot/crawl/strategy.go

```diff
@@ -87,29 +72,36 @@ func (p *PortRange) Next(addr net.Addr) (ok bool) {
 	return
 }
 
-func (p *PortRange) nextPort() (int, bool) {
-	for {
-		p.pos++
-
-		i := p.pos << p.shift
-		if i > p.Mask {
-			return 0, false // we're done
-		}
-
-		if i&p.Mask != 0 {
-			return int(i), true
-		}
+func (p *PortRange) nextPort() (port int, ok bool) {
+	// Are we done iterating?  We're done iterating if either
+	//   (a) we've wrapped around the uint16 index; or,
+	//   (b) we've exceeded p.High.
+	if ok = p.index != 0 && p.index <= p.High; ok {
```

Comment:
Why do we check `p.index != 0`?

Reply:
That's condition `(a)`.  If `p.index` == 0, it means we wrapped around `uint16`, so we've iterated through the entire port range.